src/main.rs

```diff
@@ -247,7 +260,28 @@ impl App {
     fn next(&self, args: &ArgMatches) -> Result<()> {
         let next_handle = self
             .handle_from_args(args)
-            .or_else(|_| self.random_handle())?;
+            .or_else(|_| {
+                if args.get_flag("reverse") {
+                    let max_attemps = 100;
+                    for _i in 0..max_attemps {
+                        let handle = self.random_handle().unwrap();
+                        let clash = self.read_clash(&handle).unwrap();
+                        if clash.is_reverse() {
+                            return Ok(handle);
+                        }
+                    }
+                    Err(anyhow::Error::msg(format!("No reverse clash found after {} attempts.", max_attemps)))
+                } else {
+                    self.random_handle()
+                }
+            })?;
+        // If we have a PUBLIC_HANDLE and a --reverse flag, throw an error if the clash doesn't have reverse
+        if args.get_flag("reverse") {
+            let clash = self.read_clash(&next_handle).unwrap();
+            if !clash.is_reverse() {
+                return Err(anyhow::Error::msg("The given handle doesn't has reverse mode."))
+            }
+        }
```

Comment:
It would probably be a good idea to make the mode argument and PUBLIC_HANDLE mutually exclusive using [argument group](https://docs.rs/clap/latest/clap/struct.Arg.html#method.group) so we don't even need to worry about this edge case. Clap would then automatically print nice error messages like `error: the argument '[PUBLIC_HANDLE]' cannot be used with '--reverse'`

Reply:
I will wait a bit in case `clash next` ends up being split into two different commands.